frontend/views/containers/chatroom/LeaveChannelModal.vue

```diff
@@ -33,17 +34,22 @@ export default ({
     ButtonSubmit
   },
   computed: {
-    ...mapGetters([
-      'groupSettings'
-    ])
+    ...mapGetters(['currentChatRoomId', 'currentChatRoomState']),
+    ...mapState(['loggedIn'])
   },
   methods: {
     close () {
       this.$refs.modal.close()
     },
-    submit () {
-      this.close()
-      console.log('Todo')
+    async submit () {
+      try {
+        await sbp('gi.actions/chatroom/leave', {
+          contractID: this.currentChatRoomId,
+          data: { member: this.loggedIn.username }
+        })
+      } catch (e) {
+        console.error('LeaveChannelModal submit() error:', e)
```

Comment:
Remember - all user-initiated actions that fail must report back to the user an error visually, so that the user knows the action failed and why.

Reply:
Yeap. I missed this.
Will add this one. `this.$refs.formMsg.danger(e.message)`